dts/arm/infineon/psoc6/mpns/CY8C624ABZI_S2D44.dtsi

```diff
@@ -0,0 +1,59 @@
+/*
```

Comment:
Correct. the board should choose/use MPN and doesn't care about common part (psoc6_xx_dtsi) or package part (psoc6_xx.<package_type>_dtsi).

so MPN (manufacture part number) e.g. CY8C624ABZI_S2D44.dtsi includes psoc6_02.dtsi (common part for some die type) and psoc6_02.124-bga.dtsi (package_type)

example of using this from board you can see here: boards/arm/cy8cproto_062_4343w/cy8cproto_062_4343w.dts (refer to line#7)

Board just include MPN which currently soldered on this board. 
To decrease number of files for review the dts and soc has files only for one MPN (CY8C624ABZI_S2D44), and die family (psoc6_02)

Finally, we will provide  dts and soc for all PSoC6 MPNs (in separate PR ), you can look at how it will looks like:
https://github.com/ifyall/zephyr/tree/develop/dts/arm/infineon/psoc6/mpns
https://github.com/ifyall/zephyr/tree/develop/dts/arm/infineon/psoc6